### PR TITLE
Clean up things a bit and expose StateValue to the VM adaptor and session.

### PR DESCRIPTION
**Pull request by msmouse**
### Description

Clean things up a bit before introducing changes in the SesssionExt.

### Test Plan

Existing tests 

and replay-verify
https://github.com/aptos-labs/aptos-core/actions/runs/4198844186

### CODE REVIEW

**Review by zekun000 — APPROVED** [review]
is the fix to resource group writeset called out somewhere?

**Comment by davidiw** [comment]
> is the fix to resource group writeset called out somewhere?

Huh?

This looks like he finally merged the conversion from Move finish -> Move output -> Aptos output to Move finish -> Aptos output. Should be a decent win....

File: aptos-move/aptos-gas/src/transaction/storage.rs [hunk 1]
```diff
@@ -271,7 +271,12 @@ impl ChangeSetConfigs {
         }
     }
 
-    pub fn creation_as_modification(&self) -> bool {
+    pub fn legacy_resource_creation_as_modification(&self) -> bool {
+        // Bug fixed at gas_feature_version 3 where (non-group) resource creation was converted to
+        // modification.
+        // Modules and table items were not affected (https://github.com/aptos-labs/aptos-core/pull/4722/commits/7c5e52297e8d1a6eac67a68a804ab1ca2a0b0f37).
+        // Resource groups were not affected because they were
+        // introduced later than feature_version 3 on all networks.
```
Comment (davidiw) [inline aptos-move/aptos-gas/src/transaction/storage.rs:279]: ooooh, now I see, hahahaha

how did this get through replays?

**Review by davidiw — APPROVED** [review]

**Comment by zekun000** [comment]
@davidiw I was referring to this commit https://github.com/aptos-labs/aptos-core/pull/6652/commits/26566dd9691f2052689255d7089c59497d137b6f

configs.creation_as_modification() is a hack for history and not supposed to use for resource group afaiu

**Comment by msmouse** [comment]
> is the fix to resource group writeset called out somewhere?
it's a no-op fix -- resource groups didn't exist before feature version3, so the wrong branch was never hit

**Pull request merged**